src/Bicep.Core/Workspaces/SourceFileGrouping.cs

```diff
@@ -22,13 +26,17 @@ public class SourceFileGrouping : IArtifactFileLookup
     public SourceFileGrouping(IFileResolver fileResolver,
         Uri entryFileUri,
         ImmutableDictionary<Uri, ResultWithDiagnostic<ISourceFile>> fileResultByUri,
-        ImmutableDictionary<BicepSourceFile, ImmutableDictionary<IArtifactReferenceSyntax, Result<Uri, UriResolutionError>>> fileUriResultByArtifactReference,
+        ImmutableDictionary<BicepSourceFile, ProviderDescriptorBundle> providerDescriptorBundleBySourceFile,
+        ImmutableDictionary<BicepSourceFile, ImmutableDictionary<IArtifactReferenceSyntax, Result<Uri, UriResolutionError>>> fileUriResultByArtifactReferenceSyntax,
+        ImmutableDictionary<BicepSourceFile, ImmutableDictionary<ArtifactReference, Result<Uri, UriResolutionError>>> fileUriResultByArtifactReference,
         ImmutableDictionary<ISourceFile, ImmutableHashSet<ISourceFile>> sourceFileParentLookup)
     {
         FileResolver = fileResolver;
         EntryFileUri = entryFileUri;
         FileResultByUri = fileResultByUri;
-        FileUriResultByArtifactReference = fileUriResultByArtifactReference;
+        FileUriResultByBicepSourceFileByArtifactReferenceSyntax = fileUriResultByArtifactReferenceSyntax;
```

Comment:
The length of this property name is an indication that we need to simplify something :)

Reply:
Whether or not this level of partitioning is necessary depends on the implementation of the `NamespaceResolver` changes, so I don't quite know if this is needed or not but I rather err on the side of caution and simplify in future PRs